tests/unit/composables/useUiDesign.spec.js

```diff
@@ -0,0 +1,61 @@
+import { useUiDesign } from 'src/composables/useUiDesign.ts';
+import { setUiDesign } from 'src/services/uiDesignService.ts';
+import { beforeEach, describe, expect, it } from 'vitest';
+
+describe('Test composable: useUiDesign', () => {
+  const configuration = {
+    default: {
+      button: {
+        flat: true,
+        icon: 'icon',
+        color: 'primary',
+      },
+    },
+    custom: {
+      button: {
+        flat: false,
+      },
+      other: {
+        button: {
+          color: 'secondary',
+        },
+      },
+    },
+  };
+
+  beforeEach(() => {
+    setUiDesign(configuration);
```

Comment:
The beforeEach calls `setUiDesign` on each test, but the singleton is never reset (unlike the service test which does `vi.resetModules()`). The first test will pass, but subsequent ones will use the same singleton with a warning in the console.